presto-spi/src/main/java/com/facebook/presto/spi/function/IntermediateInputFunction.java

```diff
@@ -11,15 +11,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.facebook.presto.operator.aggregation;
+package com.facebook.presto.spi.function;
 
-import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-@Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.METHOD)
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target(METHOD)
 public @interface IntermediateInputFunction
```

Comment:
The end goal of this is to remove `getServices` from `Plugin`, which can't happen until we have a replacement for `FunctionFactory` in the SPI. @martint says it's fine to add for now (it is used by a private plugin at Facebook). We can remove it once it's removed from the engine.